### PR TITLE
Update cheerio to 1.0.0

### PR DESCRIPTION
**Pull request by bcomnes**
It changes to default behaviors, but sitedown still works the same!  Had to accommodate these in the tests.

Put into a separate PR just to keep the other one clean.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "url": "https://github.com/hypermodules/sitedown/issues"
   },
   "dependencies": {
-    "cheerio": "^0.22.0",
+    "cheerio": "^1.0.0-rc.3",
```
Comment (ungoldman) [inline package.json:13]: might be better to wait until final cheerio 1.x release but if you need this and it's blocking you from doing something else i won't stop you!

**Review by ungoldman — APPROVED** [review]
fine by me as long as release is major 👍

File: package.json (hunk 1 above)
Comment (bcomnes) [reply package.json:13]: Figured the same until I noticed that rc 3 was released 2 years ago and is on the latest npm channel.  Just trying to play catch up on transient warnings.  

I'll land this in a major, but keeping it separate will make it easy to revert for whatever reason if its determined the old version was better.

**Comment by bcomnes** [comment]
Cool ty @ungoldman !

**Comment by bcomnes** [comment]
Will kick a release when I have some time off work.

**Pull request merged**